index.bs

```diff
@@ -3049,6 +3051,14 @@ The [=remote end steps=] with |command parameters| are:
     1. Set the layout [[css2#viewport]] of the |context| to be |width| CSS pixels wide and |height| CSS pixels high.
 1. Else:
     1. Set the layout [[css2#viewport]] of |context| to the implementation-defined default layout viewport for |context|.
+1. Let |device pixel ratio parameter| be the |command parameters|["<code>devicePixelRatio</code>"] field.
+1. If |device pixel ratio parameter| is not null:
+  1. If the implementation is not able to set |device pixel ratio parameter| for |context| for any reasonm return [=error=] with [=error code=] [=unsupported operation=].
```

Comment:
You want to check everything can be set upfront and fail the command if it can't rather than making it apply some settings before failing. You also potentially want to check the value is within implementation limits e.g. a device pixel ratio of 1E10 or 1E-10 should probably fail.

Reply:
Should we write down specific limitations or is it okay to cover it via the implementation specific unsupported operation?

Reply:
I think the only limitation we should impose is that it has to be > 0. From a quick glance at the code, Gecko doesn't seem to have other specific limits on our existing pref, although I imagine we have limits elsewhere that make extreme values not work.

Reply:
I have added the range limit (float .gt 0.0) to the cddl type